queue/js/queue.js

```diff
@@ -0,0 +1,21 @@
+// Queue is an implementation of a queue using a doubly-linked list.
+class Queue {
+  constructor() {
+    this.linkedList = new DoublyLinkedList();
+  }
+
+  // Adds an element to the back of the queue
+  enqueue(element) {
+    this.linkedList.addLast(element);
+  }
+
+  // Removes the element at the front of the queue
+  dequeue() {
+    return this.linkedList.removeFirst();
+  }
+
+  // Returns the element at the front of the queue
+  peek() {
+    return this.linkedList.peek();
+  }
+}
```

Comment:
Looks good. You should add driver code too @chidiwilliams

Reply:
I'll have to implement linked list on the file too to be able to add driver code.

Reply:
Can you implemented linkedlist differently and use it here? 

If yes, we can merge this for now and you can update when that's done.

Reply:
Sure, could do that.